logs/querysample/tracing.go

```diff
@@ -0,0 +1,152 @@
+package querysample
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+	pg_query "github.com/pganalyze/pg_query_go/v4"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func urlToSample(server *state.Server, grant state.GrantLogs, sample state.PostgresQuerySample) string {
+	fp := util.FingerprintQuery(sample.Query, server.Config.FilterQueryText, -1)
+	fpBin := make([]byte, 8)
+	binary.BigEndian.PutUint64(fpBin, fp)
+
+	return fmt.Sprintf(
+		"%s/databases/%s/queries/%s/samples/%d?role=%s",
+		grant.Config.ServerURL,
+		sample.Database,
+		hex.EncodeToString(fpBin),
+		sample.OccurredAt.Unix(),
+		sample.Username,
+	)
+}
+
+func parseTags(query string) map[string]string {
+	tokens, err := pg_query.Scan(query)
+	if err != nil {
+		return nil
+	}
+
+	tags := make(map[string]string)
+	for _, token := range tokens.GetTokens() {
+		var comment string
+		if token.Token == pg_query.Token_SQL_COMMENT {
+			comment = query[token.Start+2 : token.End]
+		} else if token.Token == pg_query.Token_C_COMMENT {
+			comment = query[token.Start+2 : token.End-2]
```

Comment:
The math here is because you're turning `--comment1` into `comment1` and `/*comment2*/` into `comment2`, right? It might be good to add a... comment to that effect.